packages/core/src/services/instance/instance.service.ts

```diff
@@ -172,6 +183,15 @@ export class UniverInstanceService extends Disposable implements IUniverInstance
     __addUnit(unit: UnitModel, options?: ICreateUnitOptions): void {
         const type = unit.type;
 
+        if (this.beforeUnitAddCallbacks.size > 0) {
+            for (const callback of this.beforeUnitAddCallbacks) {
+                if (callback(unit) === false) {
+                    // if not use throw error, the createUnit returns null
+                    throw new Error('[UniverInstanceService]: cannot add unit with user callback.');
```

Comment:
@wzhudev Check if this has been handled properly.

Reply:
I don't know. It is your responsibility to check if it is handled properly.